simulator/infra/macro.h

```diff
@@ -218,4 +218,39 @@ auto test_subtraction_overflow( T_src1 src1, T_src2 src2)
     return std::make_pair( narrow_cast<T>( result), is_overflow);
 }
 
+template<typename T> static
+auto shuffle_stage( T src, uint64_t maskL, uint64_t maskR, int N)
```

Comment:
Should 'T' be used instead of `uint64_t`?
For `N`, it's better to use `size_t`.